src/SmartComponents/Drift/DriftDropDown.js

```diff
@@ -0,0 +1,52 @@
+
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import { translateDriftDropdownItems } from '../../AppConstants';
+
+import {
+    Dropdown,
+    DropdownToggle,
+    DropdownItem
+} from '@patternfly/react-core';
+import CaretDownIcon from '@patternfly/react-icons/dist/esm/icons/caret-down-icon';
+
+export const DriftDropDown = ({ fetchDriftData, selectedFilter }) => {
+
+    const intl = useIntl();
+    const dropdownItems = useRef([]);
+    const [isOpen, setIsOpen] = useState(false);
+
+    useEffect(() => {
+        dropdownItems.current = translateDriftDropdownItems(intl);
+    }, []);
+
+    const onItemClick = (item) => {
+        fetchDriftData(item);
+        setIsOpen((prevOpen) => !prevOpen);
+    };
+
+    return <Dropdown
+        toggle={
+            <DropdownToggle onToggle={setIsOpen} toggleIndicator={CaretDownIcon}>
+                {selectedFilter.description}
+            </DropdownToggle>
+        }
+        isOpen={isOpen}
+        dropdownItems={dropdownItems.current.map((item, key) => (
+            <DropdownItem
+                key={key}
+                ref={el => dropdownItems.current[key] = el}
```

Comment:
You shouldn't replace the whole item, instead store the `el` in the structure.
```suggestion
                ref={el => dropdownItems.current[key].el = el}
```

Reply:
This is not working... Thee `el` is not being stored in the item

Reply:
```javascript

    const onItemClick = (item, itemRef) => {
        fetchDriftData(item);
        setIsOpen((prevOpen) => !prevOpen);
        itemRef.focus();
    };

    return <Dropdown
        toggle={
            <DropdownToggle onToggle={setIsOpen} toggleIndicator={CaretDownIcon}>
                {selectedFilter.description}
            </DropdownToggle>
        }
        isOpen={isOpen}
        dropdownItems={dropdownItems.current.map((item, key) => (
            <DropdownItem
                key={key}
                ref={(el) => dropdownItems.current[key].el = el}
                onClick={() => onItemClick(item, dropdownItems.current[key].el)}>
                {item.description}
            </DropdownItem>
        ))}
    />;
};
```

Reply:
![image](https://user-images.githubusercontent.com/13910652/150545561-9acaec37-4743-4df9-a84f-c774d8a576fa.png)